test/e2e/storage/aws_node_down.go

```diff
@@ -0,0 +1,289 @@
+
+		test := storageClassTest{
+			name:      "default",
+			claimSize: "2Gi",
+		}
+
+		pvc := newClaim(test, ns, "default")
+		pvc, err = c.CoreV1().PersistentVolumeClaims(pvc.Namespace).Create(pvc)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(pvc).ToNot(Equal(nil))
+
+		By("Waiting for pvc to be in bound phase")
+		pvcClaims := []*v1.PersistentVolumeClaim{pvc}
+		pvs, err := framework.WaitForPVClaimBoundPhase(c, pvcClaims, framework.ClaimProvisionTimeout)
+		Expect(err).NotTo(HaveOccurred(), "Failed waiting for PVC to be bound %v", err)
+
+		By("Creating a deployment")
+		deployment, err := framework.CreateDeployment(c, int32(1), map[string]string{"test": "app"}, ns, pvcClaims, "")
+		defer c.Extensions().Deployments(ns).Delete(deployment.Name, &metav1.DeleteOptions{})
+
+		By("Get a pod from deployment")
+		podList, err := framework.GetPodsForDeployment(c, deployment)
+		Expect(podList.Items).NotTo(BeEmpty())
+		pod := podList.Items[0]
+		node1 := types.NodeName(pod.Spec.NodeName)
+
+		kid := pvs[0].Spec.PersistentVolumeSource.AWSElasticBlockStore.VolumeID
+
+		volumeID, err := awscloud.GetAWSVolumeID(kid)
+		Expect(err).NotTo(HaveOccurred(), "Error converting kube volume id %s to aws volume id with %v", kid, err)
+
+		By("Veryfing if disk is attached")
+		ec2Instance, err := findAWSInstanceByNodeName(node1, ec2Client)
+
+		if err != nil {
+			framework.Failf("Error fetching node %s from AWS with error : %v", node1, err)
+		}
+
+		oldInstaneId := aws.StringValue(ec2Instance.InstanceId)
+
+		err = waitForAWSAttachedState(oldInstaneId, volumeID, ec2Client)
+		Expect(err).NotTo(HaveOccurred(), "volume %s to attached to node %s but got %v", volumeID, node1, err)
+
+		By("Shutting down the AWS Node")
+		err = shutDownAWSNode(ec2Instance, ec2Client)
+		if err != nil {
+			framework.Failf("Error while shutting down node %s with %v", node1, err)
+		}
+
+		err = waitForAWSInstanceState(node1, ec2Client, "stopped")
+
+		if err != nil {
+			framework.Failf("Error while waiting for node %s shutdown with %v", node1, err)
+		}
+
+		By("Waiting for pod to fail over to different node")
+		node2, err := waitForPodToFailover(c, deployment, node1)
+		Expect(err).NotTo(HaveOccurred(), "Pod did not fail over to a different node")
+
+		newInstance, err := findAWSInstanceByNodeName(node2, ec2Client)
+
+		if err != nil {
+			framework.Failf("Error fetching node %s from AWS with error : %v", node2, err)
+		}
+
+		newInstanceID := aws.StringValue(newInstance.InstanceId)
+
+		By("Waiting for volume to detach from old node")
+		err = waitForAWSDetachedState(oldInstaneId, volumeID, ec2Client)
+		Expect(err).NotTo(HaveOccurred(), "Error waiting for volume %s to detach from %s", volumeID, node1)
+
+		By("Waiting for volume to attach to new Node")
+		err = waitForAWSAttachedState(newInstanceID, volumeID, ec2Client)
+		Expect(err).NotTo(HaveOccurred(), "Error waiting for volume %s to attach to %s", volumeID, node2)
+
+		By("Waiting for old node to start")
```

Comment:
@justinsb I have noticed somewhat strange behaviour while running the e2e tests against a cluster started by kops. Is kops or `instanceInitiatedShutdownBehavior` causes termination of node after shutting it down? Because I noticed that if  I stop a node created by kops, it is getting terminated automatically.

Reply:
@justinsb I am not super familiar with the way kops rolls out AWS deployment. Is it because of nodes placed in autoscaling group? If yes - will the test work as part of E2E infrastructure? Need your input.

Reply:
Ah yes, kops does run instances in AutoScalingGroups.  I think everyone does apart from Openshift

Reply:
Okay I am thinking I will simply drop the test lines that start the node and wait for node to be running. But I am wondering if that will cause test cluster to be impaired and causes flakiness of E2E test suite.

Reply:
Or we can drop the whole test case/suite... Don't feel very good about that though, given I went through all the trouble. :D

Reply:
Maybe there's a tag you can add to the name of the test such that it won't be run in "normal" e2e?  Maybe `[openshift]` ?

Reply:
The `[Features:aws]` tag will cause this e2e to run its own suite separate from other tests, AFAIK. It also causes this test to be skipped during normal CI runs.

Reply:
Thanks @jeffvance - but note this isn't the normal AWS configuration either though, so not sure we want it in `[Features:aws]` :-)

Reply:
I will delete the test and we can figure out how to add it back later.